engine/Shopware/Controllers/Backend/Base.php

```diff
@@ -194,11 +194,32 @@ public function getCustomerGroupsAction()
             'groups.taxInput as taxInput',
             'groups.mode as mode',
         ]);
-        $builder->addFilter($this->Request()->getParam('filter', []));
-        $builder->addOrderBy($this->Request()->getParam('sort', []));
 
-        $builder->setFirstResult($this->Request()->getParam('start'))
-            ->setMaxResults($this->Request()->getParam('limit'));
+        $request = $this->Request();
+        $filter = $request->getParam('filter', []);
+
+        if ($query = trim($request->getParam('query', ''))) {
```

Comment:
```suggestion
        $query = trim($request->getParam('query', ''))
        if ($query !== '') {
```
assignments in if :-1: 

you also would have implicit castings on the result, which will be used for the if check. better use explicit comparisons